packages/js/product-editor/src/contexts/validation-context/README.md

```diff
@@ -0,0 +1,80 @@
+# Validations and error handling
+
+This directory contains some components and hooks used for validations in the product editor.
+
+## Add a link to error snackbar
+
+### What happens when there is an error in the form?
+
+1- Fields registered in the validator context [will get validated](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/product-editor/src/contexts/validation-context/validation-provider.tsx#L87-L110). A field can be registered by making use of the useValidation hook.
+
+-   For instance:
+
+```javascript
+const {
+	ref: myRef,
+	error: myValidationError,
+	validate: validateMyField,
+} = useValidation <
+Product >
+( 'myfield',
+async function myFieldValidator() {
+	if ( ! myField ) {
+		return {
+			message: 'My error message',
+			context: clientId,
+		};
+	}
+},
+[ myField ] );
+```
+
+2- An object consisting of the error/validation message, the context, and the validatorId will be returned ([link](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/product-editor/src/contexts/validation-context/validation-provider.tsx#L74) ).
+
+-   The `context` contains the block Id, and the `validatorId` a unique ID for the validator specifically ( generally a prefix with the block id ).
+-   If, for instance, the name field is empty, the validation will fail and will throw an object like this:
+
+    ```javascript
+    { message: 'Product name is required.'; context: [block id]; validatorId: [prefix + block id] }
+    ```
+
+-   This is the result of the name validator and the [validatorId addition](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/product-editor/src/contexts/validation-context/validation-provider.tsx#L69).
+
+    ```javascript
+    async function nameValidator() {
+      if ( ! name || name === AUTO_DRAFT_NAME ) {
+        return {
+          message: __( 'Product name is required.', 'woocommerce' ),
+          context: clientId,
+        };
+      }
+
+      if ( name.length > 120 ) {
+        return {
+          message: __(
+            'Please enter a product name shorter than 120 characters.',
+            'woocommerce'
+          ),
+          context: clientId,
+        };
+      }
+    },
+    ```
+
+3- If an error is present on the form we will show an error snackbar with the error message. We will actually include a **show error** link if the field with the relevant error is not visible ( like on another tab ). Clicking the **show error** link will direct users to the relevant field.
+
+    - We create this link by making use of the `context` property (the block id), this makes use of the `useBlocksHelper()` hook to get the parent tab id. We can do this by making use of the `core/block-editor` store and using `getBlockParentsByBlockName` (link to relevant code).
```

Comment:
I think adding the dot in the numbered bullet above should fix this weird formatting where its in a code block.
As it would be nice if it was just a normal sub-bullet list.